### ISSUE TITLE
Crypto provider support

### ISSUE THREAD

**Issue by stevefan1999-personal** [rustls/tokio-rustls#15]
Given that https://github.com/rustls/rustls/pull/1405 is closer to completion, we need to think about crypto provider support.

We could still provide `ring` provider out of the box (I think `rustls` would in the end include `ring` as part of the default options to prevent chaos), but this would render custom crypto provider useless due to `tokio-rustls` usually being a transient dependency (for example `axum_server` uses`tokio-rustls`) and I don't think there is a way to patch the transient dependency of `axum_server` without forking it.

**Comment by djc** [comment]
I don't think it would be useless? We would make `ring` a Cargo feature that is enabled by default, and we could do the same thing in tokio-rustls (and hyper-rustls), so that you have the option of disabling the ring provider at each level of the stack.

**Comment by richarddd** [comment]
rustls has now extracted crypto provider into separate dependency. Can we do the same here? 🙂

**Comment by cpu** [comment]
I believe this has been handled w/ 0.25:

`ring` is [enabled by default](https://github.com/rustls/tokio-rustls/blob/ff32e1ec589e3a81425d0721f9595e786231683d/Cargo.toml#L21) but can be disabled. If you disable that feature and construct your `Client`/`Server` configs using the `_with_provider` fns you can provide your own non-default crypto provider.
